### PR TITLE
feat(lsg): introduce icon and pattern-list-item component

### PR DESCRIPTION
**Pull request by Alexpeschel**

### CODE REVIEW

File: src/lsg/patterns/icons/demo.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import {Icon, IconName, IconRegistry, reduce, Size as IconSize} from './index';
+import * as React from 'react';
+import styled from 'styled-components';
+
+const StyledDemoIconList = styled.ul`
+  box-sizing: border-box;
```
Comment (LKuechler) [inline src/lsg/patterns/icons/demo.tsx]: should we not also have tabs in the styles?

File: src/lsg/patterns/icons/demo.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+`;
+
+interface DemoIconsProps {
+	names: typeof IconName;
+	size: IconSize;
+}
+
+const DemoIcons = (props: DemoIconsProps) => (
+	<div>
+	<StyledDemoIconList>
```
Comment (LKuechler) [inline src/lsg/patterns/icons/demo.tsx]: this should be indented

Comment (LKuechler) [reply src/lsg/patterns/icons/demo.tsx]: or can the surrounding diff be removed?

File: src/lsg/patterns/icons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		stroke-width: 1.2px;
+	}
+`;
+
+const IconRegistrySymbol: React.StatelessComponent<IconRegistrySymbolProps> = props =>
+	(
+		<symbol
+		id={`${props.id}`}
+		viewBox="0 0 24 24" >
+		{props.children}
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: this should be indented

File: src/lsg/patterns/icons/index.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		</symbol>
+	);
+
+export const IconRegistry: React.StatelessComponent<IconRegistryProps> = (props): JSX.Element =>
+(
+	<StyledIconRegistry>
+	{reduce(props.names, (name, e) => {
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: indentation :)

File: src/lsg/patterns/icons/index.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+function getIconRef(name: string): string {
+	return `#${name}`;
+}
+
+export const Icon: React.StatelessComponent<IconProps> = (props): JSX.Element => {
+	const icon = typeof props.name === 'number' ? IconName[props.name] : null;
+	return (
+		<StyledIcon className={props.className} iconColor={props.color} size={props.size}>
+		{icon !== null && <use xlinkHref={getIconRef(icon)} />}
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: indentation :)

File: src/lsg/patterns/icons/index.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+	})}
+	</StyledIconRegistry>
+);
+
+function getIconRef(name: string): string {
+	return `#${name}`;
+}
+
+export const Icon: React.StatelessComponent<IconProps> = (props): JSX.Element => {
+	const icon = typeof props.name === 'number' ? IconName[props.name] : null;
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: Why the extra check for number?

File: src/lsg/patterns/icons/index.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		<StyledIcon className={props.className} iconColor={props.color} size={props.size}>
+		{icon !== null && <use xlinkHref={getIconRef(icon)} />}
+		</StyledIcon>
+	);
+};
+
+export function reduce(e: typeof IconName, cb: (name: string, e: number) => JSX.Element[]): JSX.Element[] {
+	const results = [];
+
+	for (const n in e) {
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: Complete Names would be much better

File: src/lsg/patterns/icons/index.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+		if (isNaN(Number(n))) {
+		results.push(...cb(n, Number(e[n])));
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: indentation :)

File: src/lsg/patterns/icons/index.tsx [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+		{icon !== null && <use xlinkHref={getIconRef(icon)} />}
+		</StyledIcon>
+	);
+};
+
+export function reduce(e: typeof IconName, cb: (name: string, e: number) => JSX.Element[]): JSX.Element[] {
+	const results = [];
+
+	for (const n in e) {
+		if (isNaN(Number(n))) {
```
Comment (LKuechler) [inline src/lsg/patterns/icons/index.tsx]: should this be `Number.isNaN`?

Comment (LKuechler) [reply src/lsg/patterns/icons/index.tsx]: Ok the browser support is better this way

File: src/lsg/patterns/icons/index.tsx (hunk 6 above)
Comment (Alexpeschel) [reply src/lsg/patterns/icons/index.tsx]: otherwise props.name is falsy

**Pull request merged**